feat(theme-docs): support contents hidden in menu

NozomuIkuta opened the pull request:
<!--- Provide a general summary of your changes in the title above -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)


## Description
<!--- Describe your changes in detail -->
<!--- Why is this change required? What problem does it solve? -->
<!--- If it resolves an open issue, please link to the issue here. For example "Resolves: #137" -->

Resolves #626.
Implementation is inspired by https://github.com/nuxt/content/commit/f67ec7de49db6d49692d836e1d09f944e33da52f.

## Checklist:
<!--- Put an `x` in all the boxes that apply. -->
<!--- If your change requires a documentation PR, please link it appropriately -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why)

Currently, there is no unit tests for theme-docs.
Instead, I tested in the following way:

~- Run `yarn create nuxt-content-docs test-hidden-in-menu`~
~- Link `yarn link @nuxt/content-theme-docs` in my local~
~- Run `yarn run dev`~
~- See Setup page is shown in the menu~
~- Set `hiddenInMenu: false` to `setup.md`~
~- See Setup page is hidden in the menu~
~- See Setup page is searchable in the search box (I didn't test with Algolia search)~

_Updated Test:_

- Run `yarn create nuxt-content-docs test-hidden-in-menu`
- Run `yarn link @nuxt/content-theme-docs` in my local
- Create markdown files:
  - `index.md` (`position`: 1)
  - `2.md` (`position`: 2)
  - `3.md` (`position`: 3, `hiddenInMenu`: true)
  - `4.md` (`position`: 4)
  - `5.md` (`position`: 5)
- Run `yarn run dev`
- Check that `3.md` is hidden in the sidebar
- Check that `3.md` is accessible via the search box in the header
- Check that both links (prev, next) are not shown in `3.md`
- Check that `4.md` is shown as next link in `2.md`, instead of `3.md` (being consistent with what the sidebar shows)
- Check that `2.md` is shown as prev link in `4.md`, instead of `3.md` (being consistent with what the sidebar shows)
- Check that only next link is shown in `index.md` (existing behavior is not influenced)
- Check that only prev link is shown in `5.md` (existing behavior is not influenced)

Atinux commented:
Thanks for the pr @NozomuIkuta 

I think we should think of having a `sidebar` property in the page directly (I think `menuTitle` was wrong naming).

```yml
sidebar:
 - title: 'My sidebar title' # title by default
 - hidden: true # false by default
```

WDYT @benjamincanac ?

NozomuIkuta commented:
@Atinux 

Thank you for your suggestion.

I like consistent word usage 😀 
Currently, we use 3 words: menu, sidebar, and navigation (i.e. `AppNav`).

By the way, I missed one point when I created the PR. 
If a document is hidden in the sidebar, the document is expected to be hidden in prev/next link section as well, doesn't it?
In other words, positions of links in the sidebar and those in prev/next link section are expected to correspond, aren't they?

If my thought is correct, we should let theme-docs ignore, modify or even delete `position`s of hidden documents, otherwise the result of `sortBy()` includes hidden documents, and makes the 2 parts of UI inconsistent.

IMHO, it might be a little bit unintuitive to associate `sidebar.hidden` with the behavior of prev/next link section, especially for beginners. Adding explanation in documentation may be enough for that, though.

NozomuIkuta commented on packages/theme-docs/src/index.js:
```diff
@@ -19,14 +19,18 @@ function themeModule () {
 
     document.to = `${dir}/${slug}`
   })
+  // Configure content after each hook (hidden content)
+  hook('content:file:beforeInsert', (document) => {
+    document.hiddenInMenu = document.hiddenInMenu || false
+  })
```
I added a default hook, referencing the one which modifies document.to. 😀

NozomuIkuta commented on packages/theme-docs/src/pages/_.vue:
```diff
@@ -52,11 +52,17 @@ export default {
-      .sortBy('position', 'asc')
-      .surround(document.slug, { before: 1, after: 1 })
-      .fetch()
+    let prev
+    let next
+
+    if (!document.hiddenInMenu) {
+      [prev, next] = await $content(app.i18n.locale, { deep: true })
+        .where({ hiddenInMenu: false })
+        .only(['title', 'slug', 'to'])
+        .sortBy('position', 'asc')
+        .surround(document.slug, { before: 1, after: 1 })
+        .fetch()
+    }
```
I added a logic like below, so that sidebar content and prev-next links can be consistent.

- In unhidden content page, only unhidden pages are shown as prev-next links.
- In hidden content page, no prev-next links are shown.

NozomuIkuta commented:
@Atinux 

> By the way, I missed one point when I created the PR.
> If a document is hidden in the sidebar, the document is expected to be hidden in prev/next link section as well, doesn't it?
> In other words, positions of links in the sidebar and those in prev/next link section are expected to correspond, aren't they?

I fixed this point, and I updated "how I tested this PR" in the PR message.
Please take a look at that part.

I believe basic business logic has been completed.
I will change the property name `hiddenInMenu` as needed, according to the result of discussion among authors and people. 🙋‍♂️

NozomuIkuta commented:
@Atinux @benjamincanac Sorry to bother you but any updates?

Tahul commented:
Hey @NozomuIkuta :)

```
---
navigation: false
---
```

This will achieve the exact same behavior in @nuxt/content v2 / Docus!

Docus should become the replacement for the content-theme-docs!
